.github/workflows/publish-js-library.yml

```diff
@@ -0,0 +1,42 @@
+name: Build and publish JS library
```

Comment:
we have to bear in mind never have image build changes and JS library changes in the same PR as `publish.yaml` commits smart contract interfaces into the JS library.